patch/patch.go

```diff
@@ -92,3 +92,25 @@ func getLastAppliedConfig(obj *unstructured.Unstructured) []byte {
 	}
 	return []byte(annotations[v1.LastAppliedConfigAnnotation])
 }
+
+func TwoWay(src, tgt *unstructured.Unstructured, strategic bool) (_ []byte, err error) {
```

Comment:
The current behavior only created 3-way merge patches, essentially requiring the "original" state in addition to "modified" and "current" in order to properly deal with nulled fields. Without the original, the 3-way patch will be wrong. Hence the name of this -- now with a comment string -- but there's also now a TODO in there for better integrating both into the constructor, which might bend me back to multiple impls of a common interface.

Reply:
I decided to punt on the integration for now, since the patch tests expect 3-way merge patches for unregistered objects, so moved the 2-way logic to the only place that needs it.